### ISSUE TITLE
Can't find config file in 

### ISSUE THREAD

**Issue by geira** [adrienverge/yamllint#212]
The docs state that 

> If -c is not provided, yamllint will look for a configuration file in the following locations (by order of preference): 
>    [...]
>    ~/.config/yamllint/config

but that doesn't seem to work:

```
$ yamllint -c ~/.config/yamllint/config example.yml 
$ yamllint  example.yml 
example.yml
  3:12      error    too many spaces after colon  (colons)
  4:12      error    too many spaces after colon  (colons)
  5:12      error    too many spaces after colon  (colons)
  6:12      error    too many spaces after colon  (colons)
  7:12      error    too many spaces after colon  (colons)
  8:12      error    too many spaces after colon  (colons)
  10:12     error    too many spaces after colon  (colons)
  11:12     error    too many spaces after colon  (colons)
  12:12     error    too many spaces after colon  (colons)
  13:8      error    trailing spaces  (trailing-spaces)
```

Using yamllint 1.2.1 as installed via official Ubuntu Xenial repo.

**Comment by adrienverge** [comment]
Hey,

If you don't give the contents of your config file, nor your example YAML file to lint, it will be hard to help you.

By the way, yamllint 1.2.1 was released in March 2016: it suggest upgrading. You can ping the Ubuntu maintainer, or use `pip3 install --user --upgrade yamllint`.

**Comment by geira** [comment]
No, the contents of both files are totally irrelevant. This is a problem with locating the local config file. As you can see the tests pass without error when I specify the config file path using the "-c" flag. But yamllint fails to find it by itself, something it should be able to do if it it is in the correct place.

Admittedly the documentation is somewhat confusing here, since it is not clear whether "location" means "the directory to put the file in" or "the full path to the file":

> If -c is not provided, yamllint will look for a configuration file in the following locations (by order of preference):
>
>    .yamllint, .yamllint.yaml or .yamllint.yml in the current working directory
>    $XDG_CONFIG_HOME/yamllint/config
>    ~/.config/yamllint/config

Since I could not figure out if `~/.config/yamllint/config` was the name of the file itself or the directory to put it in, I tried both. Neither seem to be working. It is also not clear what the filename should be if `config` is a directory – surely it not considered good practice to put hidden files inside a directory that is already hidden?

So to sum up, no matter how I interpret the documentation, I cannot find a path that will make yamllint find the config file in my home dir. For the record, I have tried all the following:

- ~/.config/yamllint/config
- ~/.config/yamllint/config/yamllint
- ~/.config/yamllint/config/.yamllint

The only thing that *does* work is to put `.yamllint` in the working directory, but that is not a viable general solution.

For the record I did not try the latest version via pip, but I skimmed through all the closed issues and forks without finding any reference to similar bugs. With many servers to administer I prefer to use debian packages before source repos whenever possible.

**Comment by zpieslak** [comment]
Looking at the code the file should be named: ~/.config/yamllint/config. [Reference](https://github.com/adrienverge/yamllint/blob/85ccd625a382423edd988fcb796c4dd5897886fb/yamllint/cli.py#L174)
